### PR TITLE
Un-hide dash stuff and update README.

### PR DESCRIPTION
**Pull request by jskress**
### Description

This change un-hides the `dash` support from the help and documents it in the README.

Connected to https://github.com/rstudio/connect/issues/16876

### Testing Notes / Validation Steps

- [ ] `dash` support will now show in the various help output.
- [ ] Proofread the README changes.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -250,12 +255,80 @@ rsconnect write-manifest notebook my-notebook.ipynb
...
+### API/Application Deployment Options
+
+There are a variety of options available to you when deploying a Python WSGi-style
+API or a Dash-style application.  All options below apply equally to `api` and `dash`
+sub-commands.
+
+#### Including Extra Files
+
+You can include extra files in the deployment bundle to make them available when your
+notebook is run by the RStudio Connect server. Just specify them on the command line
```
Comment (slopp) [inline README.md]: s notebook with API / app

File: README.md [hunk 2]
```diff
@@ -250,12 +255,80 @@ rsconnect write-manifest notebook my-notebook.ipynb
 
 > **Note:** manifests for static (pre-rendered) notebooks cannot be created.
 
+### API/Application Deployment Options
+
+There are a variety of options available to you when deploying a Python WSGi-style
+API or a Dash-style application.  All options below apply equally to `api` and `dash`
```
Comment (slopp) [inline README.md]: I'm not sure I know what "Dash-style" means. flask-style or flask based make sense though.

File: README.md (hunk 1 above)
Comment (slopp) [reply README.md]: this occurs in a few places

File: README.md [hunk 3]
```diff
@@ -250,12 +255,80 @@ rsconnect write-manifest notebook my-notebook.ipynb
...
+
+If there is no `requirements.txt` file, the package dependencies will be determined
+from the current Python environment, or from an alternative Python executable specified
+via the `--python` option or via the `RETICULATE_PYTHON` environment variable:
+
+```bash
+rsconnect deploy api --python /path/to/python my-api/
+```
+
+You can see the packages list that will be included by running `pip freeze` yourself,
```
Comment (slopp) [inline README.md]: remove the Jupyter Notebook reference

File: README.md (hunk 1 above)
Comment (jskress) [reply README.md]: Fixed.

File: README.md (hunk 2 above)
Comment (jskress) [reply README.md]: Dropped "-style"

**Comment by jskress** [comment]
@slopp I've made the changes you noted so are ready for re-review.

As far as Conda, yes that's the intent.  I'm waiting to see how that falls out but either way, that will be a new PR.

I'll add to `NEWS.md` in a separate PR.  We need to decide what version we want to call this. I think enough has changed that we should go with `1.4.0`.

**Pull request merged**